src/szk.rs

```diff
@@ -0,0 +1,469 @@
+//! Szk is a wrapper class for the FLP proof system (accessible through the `Type` trait)
+//! that is designed to accept inputs that are already secret-shared between two aggregators
+//! and returns proofs that are also shared between the two aggregators. The underlying FLP
+//! accepts inputs that are vectors over a finite field, and its random coins and proofs are also
+//! vectors over the same field. In contrast, only the leader's SZK proof share contains any finite
+//! field elements, and the initial coins are bytestrings (`Seed`s). The wrapper class defined
+//! here uses an Xof (to be modeled as a random oracle) to sample coins and the helper's proof share,
+//! following the strategy of Prio3.
+
+#[cfg(test)]
+use crate::codec::{CodecError, Encode};
+use crate::field::FftFriendlyFieldElement;
+#[cfg(test)]
+use crate::field::FieldElement;
+#[cfg(test)]
+use crate::flp::{FlpError, Type};
+#[cfg(feature = "experimental")]
+#[cfg(test)]
+use crate::prng::{Prng, PrngError};
+use crate::vdaf::xof::Seed;
+#[cfg(test)]
+use crate::vdaf::xof::{IntoFieldVec, Xof, XofTurboShake128};
+#[cfg(test)]
+use std::borrow::Cow;
+#[cfg(test)]
+use std::marker::PhantomData;
+///Domain separation tags
+#[cfg(test)]
+const DST_PROVE_RANDOMNESS: u16 = 0;
+#[cfg(test)]
+const DST_PROOF_SHARE: u16 = 1;
+#[cfg(test)]
+const DST_QUERY_RANDOMNESS: u16 = 2;
+#[cfg(test)]
+const DST_JOINT_RAND_SEED: u16 = 3;
+#[cfg(test)]
+const DST_JOINT_RAND_PART: u16 = 4;
+#[cfg(test)]
+const DST_JOINT_RANDOMNESS: u16 = 5;
+// I don't know how to access the crate version number .
+#[cfg(test)]
+const VERSION: u8 = 8;
```

Comment:
I do mean the latter. There's already a constant defined in the VDAF module, which it would be nice to not duplicate, but I couldn't figure out how to import it.